### ISSUE TITLE
Full Log Ingest fails at Session Summarization phase

### ISSUE THREAD

**Issue by fgreg** [mudrod/mudrod#160]
Got a NullPointerException during full log ingest.

```
2017-05-03 15:03:32,931 INFO  discoveryengine.WeblogDiscoveryEngine (WeblogDiscoveryEngine.java:<init>(51)) - Started Mudrod Weblog Discovery Engine.
2017-05-03 15:03:32,931 INFO  discoveryengine.WeblogDiscoveryEngine (WeblogDiscoveryEngine.java:preprocess(98)) - Starting Web log preprocessing.
2017-05-03 15:03:32,932 INFO  discoveryengine.WeblogDiscoveryEngine (WeblogDiscoveryEngine.java:preprocess(106)) - Processing logs dated 201611.gz
2017-05-03 15:03:32,937 INFO  pre.ImportLogFile (ImportLogFile.java:execute(74)) - Starting Log Import 201611.gz
2017-05-03 15:03:33,677 INFO  util.Version (Version.java:logVersion(108)) - Elasticsearch Hadoop v5.2.0 [d85a257f9f]
2017-05-03 15:03:34,044 INFO  rdd.EsRDDWriter (RestService.java:createWriter(562)) - Writing to [podaaclog201611.gz/rawhttp]
2017-05-03 15:11:54,594 INFO  rdd.EsRDDWriter (RestService.java:createWriter(562)) - Writing to [podaaclog201611.gz/rawftp]
2017-05-03 15:15:02,009 INFO  pre.ImportLogFile (ImportLogFile.java:execute(78)) - Log Import complete. Time elapsed 689 seconds
2017-05-03 15:15:02,015 INFO  pre.CrawlerDetection (CrawlerDetection.java:execute(93)) - Starting Crawler detection rawhttp.
2017-05-03 15:15:04,370 INFO  pre.CrawlerDetection (CrawlerDetection.java:checkByRateInParallel(159)) - Original User count: 6365
[Stage 7:>                                                         (0 + 0) / 96]2017-05-03 15:19:10,729 INFO  pre.CrawlerDetection (CrawlerDetection.java:checkByRateInParallel(177)) - User count: 6317
2017-05-03 15:19:10,734 INFO  pre.CrawlerDetection (CrawlerDetection.java:execute(102)) - Crawler detection complete. Time elapsed 248 seconds
2017-05-03 15:19:10,749 INFO  pre.SessionGenerator (SessionGenerator.java:execute(66)) - Starting Session Generation.
2017-05-03 15:28:41,882 INFO  pre.SessionGenerator (SessionGenerator.java:genSessionByRefererInParallel(224)) - Initial Session count: 6084397
[Stage 10:=====================>                                  (36 + 4) / 96][Stage 10:======================>                                 (38 + 4) / 96]2017-05-03 15:46:33,816 INFO  pre.SessionGenerator (SessionGenerator.java:execute(71)) - Session generating complete. Time elapsed 1643 seconds.
2017-05-03 15:46:33,879 INFO  pre.SessionStatistic (SessionStatistic.java:execute(66)) - Starting Session Summarization.
[Stage 10:========================>                               (42 + 4) / 96][Stage 10:==========================>                             (45 + 4) / 96]2017-05-03 15:46:44,796 ERROR executor.Executor (Logging.scala:logError(91)) - Exception in task 47.0 in stage 10.0 (TID 445)
java.lang.NullPointerException
	at java.io.Reader.<init>(Reader.java:78)
	at java.io.InputStreamReader.<init>(InputStreamReader.java:72)
	at org.apache.commons.io.IOUtils.copy(IOUtils.java:1407)
	at org.apache.commons.io.IOUtils.copy(IOUtils.java:1433)
	at org.apache.commons.io.IOUtils.toString(IOUtils.java:585)
	at org.apache.commons.io.IOUtils.toString(IOUtils.java:563)
	at gov.nasa.jpl.mudrod.discoveryengine.MudrodAbstract.initMudrod(MudrodAbstract.java:71)
	at gov.nasa.jpl.mudrod.discoveryengine.MudrodAbstract.<init>(MudrodAbstract.java:56)
	at gov.nasa.jpl.mudrod.weblog.structure.RequestUrl.<init>(RequestUrl.java:43)
	at gov.nasa.jpl.mudrod.weblog.pre.SessionStatistic.processSession(SessionStatistic.java:236)
	at gov.nasa.jpl.mudrod.weblog.pre.SessionStatistic$2.call(SessionStatistic.java:157)
	at gov.nasa.jpl.mudrod.weblog.pre.SessionStatistic$2.call(SessionStatistic.java:147)
	at org.apache.spark.api.java.JavaRDDLike$$anonfun$fn$4$1.apply(JavaRDDLike.scala:153)
	at org.apache.spark.api.java.JavaRDDLike$$anonfun$fn$4$1.apply(JavaRDDLike.scala:153)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1$$anonfun$apply$23.apply(RDD.scala:796)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1$$anonfun$apply$23.apply(RDD.scala:796)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:99)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:282)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by fgreg** [comment]
I think this is a Spark issue.

The Spark workers don't have access to the full classpath of the application that launches the Spark job. So when trying to load this classpath resource, it won't find it.

**Comment by Yongyao** [comment]
@fgreg Is this problem solved. I saw you had passed this phase from the logs you just posted. https://github.com/mudrod/mudrod/issues/162

**Comment by fgreg** [comment]
@Yongyao Yes it's fixed with #161 and included in PR https://github.com/mudrod/mudrod/pull/159 . I wasn't going to close it until the PR is accepted and merged to master.
